libs/ledger-live-common/src/wallet-api/react.ts

```diff
@@ -760,3 +768,218 @@ export enum ExchangeType {
   SELL = 0x01,
   FUND = 0x02,
 }
+
+export interface Categories {
+  manifests: AppManifest[];
+  categories: string[];
+  manifestsByCategories: Map<string, AppManifest[]>;
+  selected: string;
+  setSelected: (val: string) => void;
+  reset: () => void;
+}
+
+export function useCategories(): Categories {
+  const manifestsSearchable = useManifests({ visibility: "searchable" });
+  const manifestsCompleted = useManifests({ visibility: "complete" });
+  const { categories, manifestsByCategories } =
+    useCategoriesRaw(manifestsCompleted);
+  const [selected, setSelected] = useState(DISCOVER_INITIAL_CATEGORY);
+
+  const reset = useCallback(() => {
+    setSelected(DISCOVER_INITIAL_CATEGORY);
+  }, []);
+
+  const manifests = useMemo(
+    () => [...manifestsSearchable, ...manifestsCompleted],
+    [manifestsSearchable, manifestsCompleted]
+  );
+
+  return {
+    manifests,
+    categories,
+    manifestsByCategories,
+    selected,
+    setSelected,
+    reset,
+  };
```

Comment:
We might want to take a look at the usage of the hooks and if we need to add a `useMemo` on the returned value